### PR TITLE
FIX: Prevent audio from playing on top of each other CAUSING IT TO BE LOUD

### PR DESCRIPTION
**Pull request by nattsw**
/t/56762

Debounces the playing of audio notifications.

We also have to think about what we want to do when audio playing fails because the user has not interacted with the app due to [an autoplay policy browsers have](https://developer.chrome.com/blog/autoplay/#audiovideo-elements).

I don't have tests for this because I don't think it's possible to unit test this without sufficiently mocking the plugin API and we seem to be against mocking.

### CODE REVIEW

**Review by martin-brennan — APPROVED** [review]
Very nice, love the `const`s as well!

File: assets/javascripts/discourse/initializers/chat-notification-sounds.js [hunk 1]
```diff
@@ -13,13 +19,33 @@ export default {
...
+          "User needs to interact with DOM before we can play notification sounds"
+        );
+      });
+    }
+
+    function playAudioWithDebounce(user) {
+      if (canPlay) {
+        canPlay = false;
+
+        setTimeout(() => {
```
Comment (markvanlan) [inline assets/javascripts/discourse/initializers/chat-notification-sounds.js:38]: Can you use [debounce](https://api.emberjs.com/ember/release/functions/@ember%2Frunloop/debounce) for this?

Comment (martin-brennan) [reply assets/javascripts/discourse/initializers/chat-notification-sounds.js:38]: Should be able to, this is just what Mattermost is doing in their source, I linked that code in our internal chat :)

Comment (nattsw) [reply assets/javascripts/discourse/initializers/chat-notification-sounds.js:38]: https://github.com/discourse/discourse-chat/pull/470

**Pull request merged**